gobblin-data-management/src/main/java/gobblin/data/management/policy/AbstractTimeBasedSelectionPolicy.java

```diff
@@ -16,31 +16,38 @@
 import java.util.List;
 import java.util.Properties;
 
-import com.google.common.base.Predicate;
-import com.google.common.collect.Collections2;
-import com.google.common.collect.Lists;
-
 import org.joda.time.Period;
 import org.joda.time.format.PeriodFormatter;
 import org.joda.time.format.PeriodFormatterBuilder;
 
+import com.google.common.base.Preconditions;
+import com.google.common.base.Predicate;
+import com.google.common.collect.Collections2;
+import com.google.common.collect.Lists;
+import com.typesafe.config.Config;
+import com.typesafe.config.ConfigFactory;
+
 import gobblin.data.management.version.FileSystemDatasetVersion;
 import gobblin.data.management.version.TimestampedDatasetVersion;
 
 
 /**
- * An implementation of {@link VersionSelectionPolicy} which returns versions that are newer than a certain time.
+ * An implementation of {@link VersionSelectionPolicy} which returns versions that satisfy a {@link Predicate}
+ * implemented by subclasses {@link #getSelectionPredicate()} method
  */
-public class TimeBasedSelectionPolicy implements VersionSelectionPolicy<TimestampedDatasetVersion> {
+public abstract class AbstractTimeBasedSelectionPolicy implements VersionSelectionPolicy<TimestampedDatasetVersion> {
 
-  public static final String TIME_BASED_SELECTION_LOOK_BACK_TIME_KEY = "gobblin.time.based.selection.lookback.time";
-  public static final String DEFAULT_LOOK_BACK_TIME = "7d";
+  public static final String TIME_BASED_SELECTION_LOOK_BACK_TIME_KEY = "selection.timeBased.lookbackTime";
```

Comment:
I think this property was incorrectly named. Since the selection policies are is new, I do not think we need to worry about this old key. If there are any distcp jobs using the old key I will change the job to use new key.